src/neuroglancer/sliceview/vector_graphics/vector_graphics_line_renderlayer.ts

```diff
@@ -0,0 +1,236 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ChunkState} from 'neuroglancer/chunk_manager/base';
+import {SliceView} from 'neuroglancer/sliceview/frontend';
+import {VectorGraphicsSourceOptions} from 'neuroglancer/sliceview/vector_graphics/base';
+import {MultiscaleVectorGraphicsChunkSource, RenderLayer as GenericVectorGraphicsRenderLayer, VectorGraphicsChunkSource} from 'neuroglancer/sliceview/vector_graphics/frontend';
+import {TrackableAlphaValue, trackableAlphaValue} from 'neuroglancer/trackable_alpha';
+import {TrackableFiniteFloat, trackableFiniteFloat} from 'neuroglancer/trackable_finite_float';
+import {mat4, vec3} from 'neuroglancer/util/geom';
+import {Buffer} from 'neuroglancer/webgl/buffer';
+import {GL_ARRAY_BUFFER, GL_FLOAT} from 'neuroglancer/webgl/constants';
+import {makeTrackableFragmentMain, makeWatchableShaderError, TrackableFragmentMain} from 'neuroglancer/webgl/dynamic_shader';
+import {ShaderBuilder, ShaderProgram} from 'neuroglancer/webgl/shader';
+
+export const FRAGMENT_MAIN_START =
+    '//NEUROGLANCER_VECTORGRAPHICS_LINE_RENDERLAYER_FRAGMENT_MAIN_START';
+
+const DEFAULT_FRAGMENT_MAIN = `void main() {  
+vec3 color = vec3(0,1,0);
+  
+  float distance = length(vNormal);
```

Comment:
While this does provide a very simple and convenient way for the user to specify the color and the amount of feathering, I am concerned that this exposes a lot of implementation details into the user-customizable fragment shader code (which becomes part of the JSON state), and that would make it very difficult to change the implementation in the future without breaking backwards compatibility.

I see two possible ways to resolve this:
1. Abstract all of the logic here into a drawLine function that takes a feather and color parameter.  Then the default fragment shader just becomes:

drawLine(/*color=*/vec3(0,1,0), /*feather=*/0.5);

2. Alternatively, instead of using a text field, you could just expose separate UI elements for specifying the amount of feathering and color.

If in the future there might be more ways to customize the line rendering, e.g. based on arbitrary attributes associated with each line or line vertex, then the first option is probably better.